src/backend/BackendRequest.cpp

```diff
@@ -134,6 +135,58 @@ void BackendRequest::configureSwiftParams() {
     _current.reset(new Uri(signed_url));
 }
 
+  //------------------------------------------------------------------------------
+  // Configure request for Reva.
+  //------------------------------------------------------------------------------
+  void BackendRequest::configureRevaParams() 
+  {
+    std::string uri = getOriginalUri()->getString();
+
+    if (_request_type != "COPY") {
+      std::string token = _params.getRevaToken(uri);
+      if (token != ""){
+      _headers_field.emplace_back("X-Access-Token", token);
+      }
+    }
+    else {
+      reva::CredentialMap cmap;
+      _params.getRevaCredentialMap(cmap);
+
+      //In pull mode : Dst-> Active Endpooint and Src-> Passive Endpoint
+      //IN push mode : Dst-> Passive Endpoint and Src->Active Endpoint
+      std::string active_token, passive_token;
+
+      //In pull mode original uri = destination uri
+      if(_params.getCopyMode() == Davix::CopyMode::Pull){
+          for (reva::CredentialMap::iterator itr = cmap.begin(); itr != cmap.end(); ++itr) {
```

Comment:
Is this an iteration over the whole map?
Wouldn't that mean the `active_token` and `passive_token` values depend only on the last element in the map?

Shouldn't there be a filtering by `uri`?

Reply:
This credMap is supposed to live only in the context of that transfer. So we only have two token at any time. 
If this is not destroyed when that transfer job completes then we would have to find an alternative.

Reply:
I understand. If the map is guaranteed to only have 2 elements in it, it's ok.

Making sure I have to have the correct rationale:
- `active_token` / `passive_token` naming is influenced primarily by 3rd push/pull role (E.g.: destination is active for 3rd pull / source is active for 3rd push)

Reply:
Yes, that is correct. The names are borrowed from a draft RFC on HTTP-TPC.